compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -534,73 +561,69 @@ static void PrintStub(
     Printer* p, StubType type) {
   const std::string service_name = service->name();
   (*vars)["service_name"] = service_name;
-  (*vars)["abstract_name"] = service_name + "ImplBase";
   std::string stub_name = service_name;
-  std::string client_name = service_name;
   std::string stub_base_class_name = "AbstractStub";
   CallType call_type;
   bool impl_base = false;
   bool interface = false;
   switch (type) {
+    case ASYNC_INTERFACE:
+      call_type = ASYNC_CALL;
+      interface = true;
+      impl_base = true;
```

Comment:
This is pretty confusing. It seems you have variations of `(impl_base && !interface)` lots of places and no place that uses `impl_base` by itself without also checking `interface`. What benefits by saying both the async interface and abstract class are impl_base? It seems they should share virtually no code/logic. This seems likely a hold-over from previously when both had all the methods.

Reply:
You're correct, fixed.